src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilerDiagnosticAnalyzer.cs

```diff
@@ -43,5 +43,68 @@ public sealed override void Initialize(AnalysisContext context)
 
             context.RegisterCompilationAction(CompilationAnalyzer.AnalyzeCompilation);
         }
+
+        protected bool IsNotSupportedByLiveAnalysis(int code)
+        {
+            if (code == MessageProvider.ERR_FailedToCreateTempFile ||
```

Comment:
Hmm it seems odd to add concept of live analysis in compiler analyzer. As far as I know, all of these errors are NoLocation diagnostics. Can we instead just say that Live analysis is only supported for diagnostics with location in source, will that work? We don't need any changes on the compiler analyzer then.